client/src/components/styles/auxstyles.js

```diff
@@ -50,7 +50,7 @@ color: #c6d4df;
 font-size: 12px;
 left: ${props => props.left};
 top: ${props => props.top};
-display: ${props => Object.keys(props.data).length === 0 ? 'none' : 'visible'}
+display: ${props => props.data === 'true' ? 'visible' : 'none'}
```

Comment:
I believe `visible` is not a valid value for display, so it'll default to `block`, which is fine for this case. `visibility: visible` is valid on the other hand, but for your use case, display might still be best, since you don't want the hoverbox to register pointer-events when it's set to `display: none`.